### PR TITLE
Add MultiFile Input Widget and Filter Parameter

### PR DESCRIPTION
**Pull request by imikejackson**
There are some other code cleanups and fixes in there also

### CODE REVIEW

**Comment by imikejackson** [comment]
Related PR:
https://github.com/BlueQuartzSoftware/SIMPLView/pull/71
https://github.com/BlueQuartzSoftware/DREAM3D/pull/966
https://github.com/dream3d/DREAM3DReview/pull/40

**Comment by imikejackson** [comment]
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/53
https://github.com/BlueQuartzSoftware/DREAM3DCI/pull/52

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+#include "SVWidgetsLib/QtSupport/QtSFileCompleter.h"
+#include "SVWidgetsLib/QtSupport/QtSFileUtils.h"
+#include "SVWidgetsLib/Widgets/SVStyle.h"
+
+#include "ui_MultiInputFileWidget.h"
+
+static QString s_LastUsedDir;
+
+// -----------------------------------------------------------------------------
+QStringList toStringList(const std::vector<std::string>& inputFiles)
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]: These two functions should probably be in an anonymous namespace.

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+// -----------------------------------------------------------------------------
+MultiInputFileWidget::~MultiInputFileWidget() = default;
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void MultiInputFileWidget::setupGui()
+{
+  // Catch when the filter is about to execute the preflight
+  connect(getFilter(), SIGNAL(preflightAboutToExecute()), this, SLOT(beforePreflight()));
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]:
```suggestion
  connect(getFilter(), &AbstractFilter::preflightAboutToExecute, this, &MultiInputFileWidget::beforePreflight);
```

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  // Catch when the filter is finished running the preflight
+  connect(getFilter(), SIGNAL(preflightExecuted()), this, SLOT(afterPreflight()));
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]:
```suggestion
  connect(getFilter(), &AbstractFilter::preflightExecuted, this, &MultiInputFileWidget::afterPreflight);
```

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+  // Catch when the filter wants its values updated
+  connect(getFilter(), SIGNAL(updateFilterParameters(AbstractFilter*)), this, SLOT(filterNeedsInputParameters(AbstractFilter*)));
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]:
```suggestion
  connect(getFilter(), &AbstractFilter::updateFilterParameters, this, &MultiInputFileWidget::filterNeedsInputParameters);
```

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+  connect(m_Ui->selectBtn, SIGNAL(clicked()), this, SLOT(selectMultiInputFile()));
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]:
```suggestion
  connect(m_Ui->selectBtn, &QPushButton::clicked, this, &MultiInputFileWidget::selectMultiInputFile);
```

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+  createSummaryText();
+}
+
+// -----------------------------------------------------------------------------
+void MultiInputFileWidget::selectMultiInputFile()
+{
+  QString Ftype = m_FilterParameter->getFileType();
+  QString ext = m_FilterParameter->getFileExtension();
+  QString s = Ftype + QString(" Files (") + ext + QString(");;All Files(*.*)");
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]:
```suggestion
  QString s = QString("%1 Files (%2);;All Files (*.*)").arg(Ftype, ext);
```

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+#include "SIMPLib/Utilities/SIMPLDataPathValidator.h"
+
+#include "SVWidgetsLib/Core/SVWidgetsLibConstants.h"
+#include "SVWidgetsLib/QtSupport/QtSFileCompleter.h"
+#include "SVWidgetsLib/QtSupport/QtSFileUtils.h"
+#include "SVWidgetsLib/Widgets/SVStyle.h"
+
+#include "ui_MultiInputFileWidget.h"
+
+static QString s_LastUsedDir;
```
Comment (JDuffeyBQ) [inline Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp:59]: Should this be a member variable? Or is it supposed to be shared?

Comment (imikejackson) [reply Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp:59]: We usually share it between instances. I can argue it either way.

File: Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp (hunk 1 above)
Comment (imikejackson) [reply Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]: They should probably be in a utility namespace somewhere?

Comment (imikejackson) [reply Source/SVWidgetsLib/FilterParameterWidgets/MultiInputFileWidget.cpp]: moved to SIMPLib/Utilities/QtConverter.hpp

**Pull request merged**